src/JNCC.PublicWebsite.Core/Controllers/ApiControllers/PageExportApiController.cs

```diff
@@ -0,0 +1,80 @@
+﻿using JNCC.PublicWebsite.Core.Attributes.WebApi;
+using JNCC.PublicWebsite.Core.Services;
+using SEOChecker.Core.Extensions;
+using SEOChecker.Core.Extensions.UmbracoExtensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Web;
+using System.Web.Http;
+using umbraco.cms.businesslogic.property;
+using Umbraco.Core;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+using Umbraco.Web;
+using Umbraco.Web.WebApi;
+
+namespace JNCC.PublicWebsite.Core.Controllers.ApiControllers
+{
+    public sealed class PageExportApiController : UmbracoAuthorizedApiController
+	{
+		[HttpGet]
+        [ActionName("Export")]
+		public IHttpActionResult Export()
+        {
+			IContentService _contentService = Services.ContentService;
+            IUserService _userService = Services.UserService;
+            
+            var pages = _contentService.GetRootContent().SelectRecursive(c => c.Descendants());
+            var url = HttpContext.Current.Request.Url.Scheme + "://" + HttpContext.Current.Request.Url.GetHostAndPort();
+
+            var formatted = pages.Select(p => new PageExportModel(p, _userService, Umbraco, url));
+
+            CSVExportService.GenericExport(formatted.ToArray(), "Pages", true);
+
+            return InternalServerError(new Exception("There was an issue generating the report"));
```

Comment:
I presume the fact that a successful run ends the current response means that this error doesn't happen in a successful case?

Reply:
Correct